src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,14 +161,15 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            return _curlHandler.SendAsync(request, cancellationToken);
+            return _curlHandlerPipeline.SendAsync(request, cancellationToken);
```

Comment:
I would like to avoid all checks for whether logging is currently enabled except for one upfront, e.g. make this line something like:
```C#
return s_diagnosticListener.IsEnabled() ?
    _curlHandlerPipeline.SendAsync(request, cancellationToken) :
    _curlHandler.SendAsync(request, cancellationToken);
```
such that we only have the one additional static field read / bool check upfront, and then if logging wasn't enabled, we don't have to incur any of the additional costs, such as extra branches, extra function calls, extra virtual calls, etc.  We then don't have to worry quite as much about what happens in the DiagnosticHandler in subsequent PRs with regards to perf overheads.